basictracer/tracer.py

```diff
@@ -0,0 +1,60 @@
+import time
+
+from opentracing import Format, Tracer
+from opentracing import UnsupportedFormatException
+from .context import Context
+from .propagation import BinaryPropagator, TextPropagator
+from .recorder import SpanRecorder
+from .span import BasicSpan, RawSpan
+from .util import generate_id
+
+class BasicTracer(Tracer):
+
+    def __init__(self, recorder=None):
+        self.recorder = NoopRecorder() if recorder is None else recorder
+        self._binary_propagator = BinaryPropagator(self)
+        self._text_propagator = TextPropagator(self)
+        return
+
+    def start_span(self,
+            operation_name=None,
+            parent=None,
+            tags=None,
+            start_time=None):
+
+        start_time = time.time() if start_time is None else start_time
+        context = Context(span_id=generate_id())
+        raw = RawSpan(operation_name=operation_name, tags=tags,
+                context=context,
+                start_time=start_time)
+
+        if parent is None:
+            raw.context.trace_id=generate_id()
```

Comment:
this makes every new trace sampled. You'd want to introduce a Sampler interface, for "classic" behavior.